cancermuts/datasources.py

```diff
@@ -1466,54 +1462,57 @@ def _get_mobidb_disorder_predictions_assignments(self, sequence, *args, **kwargs
-                self.log.warning("More than one prediction found for MobiDB; will use the first")
+        #Initialisation for annotation
+        disorder_types=['curated-disorder-priority','derived-disorder-priority','homology-disorder-priority','prediction-disorder-priority']
+        evidence_types = {'curated-disorder-priority': 'curated',
+                        'derived-disorder-priority': 'derived',
+                        'homology-disorder-priority': 'homology',
+                        'prediction-disorder-priority': 'prediction'}
 
-            regions = data['mobidb_consensus']['disorder']['full'][0]['regions']
+        disorder_data={'curated-disorder-priority': curated_disorder_data,
+                        'derived-disorder-priority': derived_disorder_data,
+                        'homology-disorder-priority': homology_disorder_data,
+                        'prediction-disorder-priority': predicted_disorder_data}
 
-            for r in regions:
-                for i in range(r[0]-1, r[1]):
-                    try:
-                        assignments[i] = str(r[2])
-                    except IndexError:
-                        self.log.error("residue index %s not in sequence!" % i)
-                        return None
-            if None in assignments:
-                self.log.warning("Some positions were not assigned!")
-
-        elif 'predictors' in disorder_data:
-            self.log.info("full consensus not available - predictions will be used")
-
-            mobidb_lite = data['mobidb_consensus']['disorder']['predictors'][1]
-            if mobidb_lite['method'] != 'mobidb-lite':
-                self.log.error("ModiDB-lite consensus prediction is not available")
-                return None
+        assignments = [None for p in sequence.positions]
 
-            regions = mobidb_lite['regions']
-            for r in regions:
-                for i in range(r[0]-1,r[1]): # r[1]: -1 because of the 0-offset, +1 because of the [) of range, total 0
-                    try:
-                        assignments[i] = str(r[2])
-                    except IndexError:
-                        self.log.error("residue %s not in sequence!" % i)
-                        return None
-            assignments = [ 'S' if a is None else a for a in assignments ]
+        # If specified data is available, annotate sequence accordingly 
+        for dis_term in disorder_types:
+            if disorder_data[dis_term] is not None:
+                regions = disorder_data[dis_term]['regions']
+                for r in regions:
+                    for i in range(r[0]-1, r[1]): # r[1]: -1 because of the 0-offset, +1 because of the [) of range, total 0
+                        try:
+                            assignments[i] = 'Disordered, '+evidence_types[dis_term]
+                        except IndexError:
+                            self.log.error("residue index %s not in sequence!" % i)
+                            return None
```

Comment:
yes - but I would annotate *in the opposite order* (prediction first, then homology, then...)

this is because it allows us to better handles cases in which the same residue has multiple annotations (e.g. it's predicted as disordered *and* is curated as disordered) 

it's not completely clear to me if the regions can be overlapping (we haven't found any yet?) but this makes it sure that we keep the highest-priority annotation if this happens. If we annotate in inverse order, we are guaranteed to overwrite an annotation of lower priority with one of higher priority rather than the opposite